src/lib.rs

```diff
@@ -1743,6 +1778,19 @@ pub struct EntryPoint {
     pub function: Function,
 }
 
+/// Set of special types that can be optionally generated by the frontends.
+#[derive(Debug, Default)]
+#[cfg_attr(feature = "clone", derive(Clone))]
+#[cfg_attr(feature = "serialize", derive(Serialize))]
+#[cfg_attr(feature = "deserialize", derive(Deserialize))]
+#[cfg_attr(feature = "arbitrary", derive(Arbitrary))]
+pub struct SpecialTypes {
+    /// Type for `RayDesc`.
+    ray_desc: Option<Handle<Type>>,
+    /// Type for `RayIntersection`.
+    ray_intersection: Option<Handle<Type>>,
```

Comment:
Should these fields be `pub` too? Generally we want to support third-party producers and consumers of `Module`.